app/controllers/proposals_controller.rb

```diff
@@ -2,6 +2,7 @@ class ProposalsController < ApplicationController
   include TokenAuth
 
   skip_before_action :authenticate_user!, only: [:approve]
+  before_action :check_disabled_client
```

Comment:
I think we should only need to do this for `:approve` since we are using `authenticate_user!